source/fe/fe_simplex_p.cc

```diff
@@ -769,15 +768,19 @@ FE_SimplexP<dim, spacedim>::hp_line_dof_identities(
       // hard-coded and we iterate over points on the first line which begin
       // after the 3 vertex points in the complete list of unit support points
 
-      Assert(fe_p_other->degree <= 2, ExcNotImplemented());
-
       std::vector<std::pair<unsigned int, unsigned int>> identities;
 
       for (unsigned int i = 0; i < this->degree - 1; ++i)
         for (unsigned int j = 0; j < fe_p_other->degree - 1; ++j)
           if (std::fabs(this->unit_support_points[i + 3][0] -
                         fe_p_other->unit_support_points[i + 3][0]) < 1e-14)
             identities.emplace_back(i, j);
+          else
+            // If nodes are not located in the same place, we have to
+            // interpolate. This is then not handled through the current
+            // function, but via interpolation matrices that result in
+            // constraints, rather than identities.
+            ;
```

Comment:
This construct is a bit weird to my eyes. Do we do this kind of empty statement anywhere else? Are you married to it? :-)
Also, should this be an assert? Your comment could be interpreted that this function is not usable in that case.

Reply:
This is the function that finds identities between DoFs on a common face between neighboring elements (say, the middle nodes of adjacent Q2 and Q4 elements). They then receive the same DoF number. For all cases where we don't have an identity, we deal with the face via hanging node constraints (in a different function).

Regardless, I'll change text and layout of the comment.